### PR TITLE
Cleanup error reporting in crypto/

### PR DESCRIPTION
**Pull request by levitte**
Error reporting in `crypto/` is changed to make better use of the newer ERR building blocks, and to drop the old function codes everywhere.

### CODE REVIEW

File: crypto/bio/b_addr.c [hunk 1]
```diff
@@ -213,12 +213,10 @@ static int addr_strings(const BIO_ADDR *ap, int numeric,
             if (ret == EAI_SYSTEM) {
                 ERR_raise_data(ERR_LIB_SYS, get_last_socket_error(),
                                "calling getnameinfo()");
-                BIOerr(BIO_F_ADDR_STRINGS, ERR_R_SYS_LIB);
             } else
 # endif
             {
-                BIOerr(BIO_F_ADDR_STRINGS, ERR_R_SYS_LIB);
-                ERR_add_error_data(1, gai_strerror(ret));
+                ERR_raise_data(ERR_LIB_BIO, ERR_R_SYS_LIB, gai_strerror(ret));
```
Comment (slontis) [inline crypto/bio/b_addr.c:219]: Appveyor is complaining about a WCHAR * to char* conversion here..

File: crypto/bio/b_addr.c [hunk 2]
```diff
@@ -729,8 +727,8 @@ int BIO_lookup_ex(const char *host, const char *service, int lookup_type,
                 goto retry;
             }
 # endif
-            BIOerr(BIO_F_BIO_LOOKUP_EX, ERR_R_SYS_LIB);
-            ERR_add_error_data(1, gai_strerror(old_ret ? old_ret : gai_ret));
+            ERR_raise_data(ERR_LIB_BIO, ERR_R_SYS_LIB,
+                           gai_strerror(old_ret ? old_ret : gai_ret));
```
Comment (slontis) [inline crypto/bio/b_addr.c:731]: And here also..

File: crypto/asn1/a_object.c [hunk 3]
```diff
@@ -116,7 +116,7 @@ int a2d_ASN1_OBJECT(unsigned char *out, int olen, const char *buf, int num)
         }
         if (len == 0) {
             if ((first < 2) && (l >= 40)) {
-                ASN1err(ASN1_F_A2D_ASN1_OBJECT,
+                ERR_raise(ERR_LIB_ASN1,
                         ASN1_R_SECOND_NUMBER_TOO_LARGE);
```
Comment (slontis) [inline crypto/asn1/a_object.c:120]: indent

File: crypto/asn1/a_sign.c [hunk 4]
```diff
@@ -62,18 +62,18 @@ int ASN1_sign(i2d_of_void *i2d, X509_ALGOR *algor1, X509_ALGOR *algor2,
...
         a->algorithm = OBJ_nid2obj(type->pkey_type);
         if (a->algorithm == NULL) {
-            ASN1err(ASN1_F_ASN1_SIGN, ASN1_R_UNKNOWN_OBJECT_TYPE);
+            ERR_raise(ERR_LIB_ASN1, ASN1_R_UNKNOWN_OBJECT_TYPE);
             goto err;
         }
         if (a->algorithm->length == 0) {
-            ASN1err(ASN1_F_ASN1_SIGN,
+            ERR_raise(ERR_LIB_ASN1,
                     ASN1_R_THE_ASN1_OBJECT_IDENTIFIER_IS_NOT_KNOWN_FOR_THIS_MD);
```
Comment (slontis) [inline crypto/asn1/a_sign.c:70]: indent

File: crypto/asn1/a_sign.c [hunk 5]
```diff
@@ -187,7 +187,7 @@ int ASN1_item_sign_ctx(const ASN1_ITEM *it, X509_ALGOR *algor1,
             goto err;
 
         if ((aid_len = params[0].return_size) == 0) {
-            ASN1err(ASN1_F_ASN1_ITEM_SIGN_CTX,
+            ERR_raise(ERR_LIB_ASN1,
                     ASN1_R_DIGEST_AND_KEY_TYPE_NOT_SUPPORTED);
```
Comment (slontis) [inline crypto/asn1/a_sign.c:191]: indent

File: crypto/asn1/a_sign.c [hunk 6]
```diff
@@ -243,7 +243,7 @@ int ASN1_item_sign_ctx(const ASN1_ITEM *it, X509_ALGOR *algor1,
             pkey->ameth->pkey_id;
 
         if (!OBJ_find_sigid_by_algs(&signid, EVP_MD_nid(md), pkey_id)) {
-            ASN1err(ASN1_F_ASN1_ITEM_SIGN_CTX,
+            ERR_raise(ERR_LIB_ASN1,
                     ASN1_R_DIGEST_AND_KEY_TYPE_NOT_SUPPORTED);
```
Comment (slontis) [inline crypto/asn1/a_sign.c:247]: indent

File: crypto/asn1/ameth_lib.c [hunk 7]
```diff
@@ -168,7 +168,7 @@ int EVP_PKEY_asn1_add0(const EVP_PKEY_ASN1_METHOD *ameth)
 
     tmp.pkey_id = ameth->pkey_id;
     if (sk_EVP_PKEY_ASN1_METHOD_find(app_methods, &tmp) >= 0) {
-        EVPerr(EVP_F_EVP_PKEY_ASN1_ADD0,
+        ERR_raise(ERR_LIB_EVP,
                EVP_R_PKEY_APPLICATION_ASN1_METHOD_ALREADY_REGISTERED);
```
Comment (slontis) [inline crypto/asn1/ameth_lib.c:172]: indent

File: crypto/asn1/f_int.c [hunk 8]
```diff
@@ -119,7 +119,7 @@ int a2i_ASN1_INTEGER(BIO *bp, ASN1_INTEGER *bs, char *buf, int size)
             for (n = 0; n < 2; n++) {
                 m = OPENSSL_hexchar2int(bufp[k + n]);
                 if (m < 0) {
-                    ASN1err(ASN1_F_A2I_ASN1_INTEGER,
+                    ERR_raise(ERR_LIB_ASN1,
                             ASN1_R_NON_HEX_CHARACTERS);
```
Comment (slontis) [inline crypto/asn1/f_int.c:123]: indent

File: crypto/asn1/f_string.c [hunk 9]
```diff
@@ -110,7 +110,7 @@ int a2i_ASN1_STRING(BIO *bp, ASN1_STRING *bs, char *buf, int size)
             for (n = 0; n < 2; n++) {
                 m = OPENSSL_hexchar2int(bufp[k + n]);
                 if (m < 0) {
-                    ASN1err(ASN1_F_A2I_ASN1_STRING,
+                    ERR_raise(ERR_LIB_ASN1,
                             ASN1_R_NON_HEX_CHARACTERS);
```
Comment (slontis) [inline crypto/asn1/f_string.c:114]: indent

File: crypto/asn1/p5_pbev2.c [hunk 10]
```diff
@@ -49,7 +49,7 @@ X509_ALGOR *PKCS5_pbe2_set_iv(const EVP_CIPHER *cipher, int iter,
 
     alg_nid = EVP_CIPHER_type(cipher);
     if (alg_nid == NID_undef) {
-        ASN1err(ASN1_F_PKCS5_PBE2_SET_IV,
+        ERR_raise(ERR_LIB_ASN1,
                 ASN1_R_CIPHER_HAS_NO_OBJECT_IDENTIFIER);
```
Comment (slontis) [inline crypto/asn1/p5_pbev2.c:53]: indent

File: crypto/asn1/p5_scrypt.c [hunk 11]
```diff
@@ -49,19 +49,19 @@ X509_ALGOR *PKCS5_pbe2_set_scrypt(const EVP_CIPHER *cipher,
...
     if (!cipher) {
-        ASN1err(ASN1_F_PKCS5_PBE2_SET_SCRYPT, ERR_R_PASSED_NULL_PARAMETER);
+        ERR_raise(ERR_LIB_ASN1, ERR_R_PASSED_NULL_PARAMETER);
         goto err;
     }
 
     if (EVP_PBE_scrypt(NULL, 0, NULL, 0, N, r, p, 0, NULL, 0) == 0) {
-        ASN1err(ASN1_F_PKCS5_PBE2_SET_SCRYPT,
+        ERR_raise(ERR_LIB_ASN1,
                 ASN1_R_INVALID_SCRYPT_PARAMETERS);
```
Comment (slontis) [inline crypto/asn1/p5_scrypt.c:58]: indent

File: crypto/asn1/p5_scrypt.c [hunk 12]
```diff
@@ -49,19 +49,19 @@ X509_ALGOR *PKCS5_pbe2_set_scrypt(const EVP_CIPHER *cipher,
...
         goto err;
     }
 
     alg_nid = EVP_CIPHER_type(cipher);
     if (alg_nid == NID_undef) {
-        ASN1err(ASN1_F_PKCS5_PBE2_SET_SCRYPT,
+        ERR_raise(ERR_LIB_ASN1,
                 ASN1_R_CIPHER_HAS_NO_OBJECT_IDENTIFIER);
```
Comment (slontis) [inline crypto/asn1/p5_scrypt.c:65]: indent

File: crypto/asn1/p5_scrypt.c [hunk 13]
```diff
@@ -94,7 +94,7 @@ X509_ALGOR *PKCS5_pbe2_set_scrypt(const EVP_CIPHER *cipher,
     if (EVP_CipherInit_ex(ctx, cipher, NULL, NULL, iv, 0) == 0)
         goto err;
     if (EVP_CIPHER_param_to_asn1(ctx, scheme->parameter) <= 0) {
-        ASN1err(ASN1_F_PKCS5_PBE2_SET_SCRYPT,
+        ERR_raise(ERR_LIB_ASN1,
                 ASN1_R_ERROR_SETTING_CIPHER_PARAMS);
```
Comment (slontis) [inline crypto/asn1/p5_scrypt.c:98]: insent

File: crypto/asn1/p5_scrypt.c [hunk 14]
```diff
@@ -247,7 +247,7 @@ int PKCS5_v2_scrypt_keyivgen(EVP_CIPHER_CTX *ctx, const char *pass,
         uint64_t spkeylen;
         if ((ASN1_INTEGER_get_uint64(&spkeylen, sparam->keyLength) == 0)
             || (spkeylen != keylen)) {
-            EVPerr(EVP_F_PKCS5_V2_SCRYPT_KEYIVGEN,
+            ERR_raise(ERR_LIB_EVP,
                    EVP_R_UNSUPPORTED_KEYLENGTH);
```
Comment (slontis) [inline crypto/asn1/p5_scrypt.c:251]: indent

File: crypto/asn1/p5_scrypt.c [hunk 15]
```diff
@@ -257,7 +257,7 @@ int PKCS5_v2_scrypt_keyivgen(EVP_CIPHER_CTX *ctx, const char *pass,
         || ASN1_INTEGER_get_uint64(&r, sparam->blockSize) == 0
         || ASN1_INTEGER_get_uint64(&p, sparam->parallelizationParameter) == 0
         || EVP_PBE_scrypt(NULL, 0, NULL, 0, N, r, p, 0, NULL, 0) == 0) {
-        EVPerr(EVP_F_PKCS5_V2_SCRYPT_KEYIVGEN,
+        ERR_raise(ERR_LIB_EVP,
                EVP_R_ILLEGAL_SCRYPT_PARAMETERS);
```
Comment (slontis) [inline crypto/asn1/p5_scrypt.c:261]: indent

File: crypto/asn1/tasn_dec.c [hunk 16]
```diff
@@ -172,7 +172,7 @@ static int asn1_item_embed_d2i(ASN1_VALUE **pval, const unsigned char **in,
              * template in the template itself.
              */
             if ((tag != -1) || opt) {
-                ASN1err(ASN1_F_ASN1_ITEM_EMBED_D2I,
+                ERR_raise(ERR_LIB_ASN1,
                         ASN1_R_ILLEGAL_OPTIONS_ON_ITEM_TEMPLATE);
```
Comment (slontis) [inline crypto/asn1/tasn_dec.c:176]: indent

File: crypto/asn1/tasn_dec.c [hunk 17]
```diff
@@ -463,35 +463,35 @@ static int asn1_template_ex_d2i(ASN1_VALUE **val,
...
         if (!ret) {
-            ASN1err(ASN1_F_ASN1_TEMPLATE_EX_D2I, ERR_R_NESTED_ASN1_ERROR);
+            ERR_raise(ERR_LIB_ASN1, ERR_R_NESTED_ASN1_ERROR);
             return 0;
         } else if (ret == -1)
             return -1;
         if (!cst) {
-            ASN1err(ASN1_F_ASN1_TEMPLATE_EX_D2I,
+            ERR_raise(ERR_LIB_ASN1,
                     ASN1_R_EXPLICIT_TAG_NOT_CONSTRUCTED);
```
Comment (slontis) [inline crypto/asn1/tasn_dec.c:472]: indent

File: crypto/asn1/tasn_dec.c [hunk 18]
```diff
@@ -463,35 +463,35 @@ static int asn1_template_ex_d2i(ASN1_VALUE **val,
...
                 goto err;
             }
         } else {
             /*
              * Otherwise we must hit the EXPLICIT tag end or its an error
              */
             if (len) {
-                ASN1err(ASN1_F_ASN1_TEMPLATE_EX_D2I,
+                ERR_raise(ERR_LIB_ASN1,
                         ASN1_R_EXPLICIT_LENGTH_MISMATCH);
```
Comment (slontis) [inline crypto/asn1/tasn_dec.c:495]: indent

File: crypto/asn1/tasn_dec.c [hunk 19]
```diff
@@ -581,7 +581,7 @@ static int asn1_template_noexp_d2i(ASN1_VALUE **val,
             /* See if EOC found */
             if (asn1_check_eoc(&p, len)) {
                 if (!sk_eoc) {
-                    ASN1err(ASN1_F_ASN1_TEMPLATE_NOEXP_D2I,
+                    ERR_raise(ERR_LIB_ASN1,
                             ASN1_R_UNEXPECTED_EOC);
```
Comment (slontis) [inline crypto/asn1/tasn_dec.c:585]: indent

File: crypto/asn1/tasn_dec.c [hunk 20]
```diff
@@ -593,21 +593,21 @@ static int asn1_template_noexp_d2i(ASN1_VALUE **val,
             if (!asn1_item_embed_d2i(&skfield, &p, len,
                                      ASN1_ITEM_ptr(tt->item), -1, 0, 0, ctx,
                                      depth)) {
-                ASN1err(ASN1_F_ASN1_TEMPLATE_NOEXP_D2I,
+                ERR_raise(ERR_LIB_ASN1,
                         ERR_R_NESTED_ASN1_ERROR);
```
Comment (slontis) [inline crypto/asn1/tasn_dec.c:597]: indent

File: crypto/asn1/tasn_dec.c [hunk 21]
```diff
@@ -666,19 +666,19 @@ static int asn1_d2i_ex_primitive(ASN1_VALUE **pval,
...
         unsigned char oclass;
         if (tag >= 0) {
-            ASN1err(ASN1_F_ASN1_D2I_EX_PRIMITIVE, ASN1_R_ILLEGAL_TAGGED_ANY);
+            ERR_raise(ERR_LIB_ASN1, ASN1_R_ILLEGAL_TAGGED_ANY);
             return 0;
         }
         if (opt) {
-            ASN1err(ASN1_F_ASN1_D2I_EX_PRIMITIVE,
+            ERR_raise(ERR_LIB_ASN1,
                     ASN1_R_ILLEGAL_OPTIONAL_ANY);
```
Comment (slontis) [inline crypto/asn1/tasn_dec.c:674]: indent

File: crypto/asn1/tasn_dec.c [hunk 22]
```diff
@@ -710,7 +710,7 @@ static int asn1_d2i_ex_primitive(ASN1_VALUE **pval,
         }
         /* SEQUENCE and SET must be constructed */
         else if (!cst) {
-            ASN1err(ASN1_F_ASN1_D2I_EX_PRIMITIVE,
+            ERR_raise(ERR_LIB_ASN1,
                     ASN1_R_TYPE_NOT_CONSTRUCTED);
```
Comment (slontis) [inline crypto/asn1/tasn_dec.c:714]: indent

File: crypto/asn1/tasn_dec.c [hunk 23]
```diff
@@ -859,19 +859,19 @@ static int asn1_ex_c2i(ASN1_VALUE **pval, const unsigned char *cont, int len,
...
     default:
         if (utype == V_ASN1_BMPSTRING && (len & 1)) {
-            ASN1err(ASN1_F_ASN1_EX_C2I, ASN1_R_BMPSTRING_IS_WRONG_LENGTH);
+            ERR_raise(ERR_LIB_ASN1, ASN1_R_BMPSTRING_IS_WRONG_LENGTH);
             goto err;
         }
         if (utype == V_ASN1_UNIVERSALSTRING && (len & 3)) {
-            ASN1err(ASN1_F_ASN1_EX_C2I,
+            ERR_raise(ERR_LIB_ASN1,
                     ASN1_R_UNIVERSALSTRING_IS_WRONG_LENGTH);
```
Comment (slontis) [inline crypto/asn1/tasn_dec.c:867]: indent

File: crypto/async/async.c [hunk 24]
```diff
@@ -212,7 +212,7 @@ int ASYNC_start_job(ASYNC_JOB **job, ASYNC_WAIT_CTX *wctx, int *ret,
                 /* Resume previous job */
                 if (!async_fibre_swapcontext(&ctx->dispatcher,
                         &ctx->currjob->fibrectx, 1)) {
-                    ASYNCerr(ASYNC_F_ASYNC_START_JOB,
+                    ERR_raise(ERR_LIB_ASYNC,
                              ASYNC_R_FAILED_TO_SWAP_CONTEXT);
```
Comment (slontis) [inline crypto/async/async.c:216]: indent

**Review by slontis — COMMENTED** [review]
I gave up on looking at this..
indenting for multi-lines needs to be fixed in lots of places..

Only real issue is the appveyor error.

File: crypto/bio/b_addr.c (hunk 1 above)
Comment (levitte) [reply crypto/bio/b_addr.c:219]: ???? How come it didn't before?

File: crypto/asn1/a_object.c (hunk 3 above)
Comment (levitte) [reply crypto/asn1/a_object.c]: Fixed all indents

File: crypto/bio/b_addr.c (hunk 1 above)
Comment (slontis) [reply crypto/bio/b_addr.c:219]: Probably because of ...
void ERR_add_error_data(int num, ...);

Comment (levitte) [reply crypto/bio/b_addr.c:219]: Oh, right, with ERR_add_error_data(), it was among the stdarg stuff...  so we probably treated a WCHAR* as a char*, just more silently.

I just looked at MSDN docs, and apparently, we have to use get_strerrorA() with Windows.
*Thank you Microsoft*

**Review by paulidale — APPROVED** [review]

**Comment by paulidale** [comment]
Assuming that the CI builds are good.

File: crypto/bio/b_addr.c (hunk 1 above)
Comment (levitte) [reply crypto/bio/b_addr.c:219]: Fixed, I hope

**Review by paulidale — DISMISSED** [review]

**Comment by levitte** [comment]
Appveyor issue fixed properly this time, I hope.

**Comment by levitte** [comment]
Rebased to catch up with master

**Review by paulidale — APPROVED** [review]

**Comment by levitte** [comment]
Merged

9311d0c471ca2eaa259e8c1bbbeb7c46394c7ba2 Convert all {NAME}err() in crypto/ to their corresponding ERR_raise() call
a150f8e1fcc38752fef4d7c75d765d8efc7d46d6 CRYPTO: refactor ERR_raise()+ERR_add_error_data() to ERR_raise_data()
105d01f1eb0e48ea24b430a0ac9636906a24cdbb crypto/provider_core.c: fix a couple of faulty ERR_raise_data() calls
256d41d4371720ccfe1a4fead6bd28ed5071bcdd BIO: Undefine UNICODE in b_addr.c to get POSIX declaration of gai_strerror()

**Pull request closed without merging**